packages/ckeditor5-fullscreen/src/handlers/abstracteditor.ts

```diff
@@ -393,4 +397,95 @@ export default class AbstractEditorHandler {
+	 * utilizing the empty space on the right side of the editable element.
+	 */
+	public registerFullscreenDialogPositionAdjustements(): void {
+		if ( !this._editor.plugins.has( 'Dialog' ) ) {
+			return;
+		}
+
+		const dialog = this._editor.plugins.get( 'Dialog' );
+
+		this._setNewDialogPosition();
+
+		dialog.on( 'change:isOpen', this.updateDialogPositionCallback, { priority: 'highest' } );
+	}
+
+	/**
+	 * Removes an event listener that adjusts the dialog's position in fullscreen mode.
+	 */
+	public unregisterFullscreenDialogPositionAdjustements(): void {
+		if ( !this._editor.plugins.has( 'Dialog' ) ) {
+			return;
+		}
+
+		const dialog = this._editor.plugins.get( 'Dialog' );
+		const dialogView = dialog.view;
+
+		if ( dialogView && dialogView.position === null ) {
+			dialogView.position = DialogViewPosition.EDITOR_TOP_SIDE;
+		}
+
+		if ( dialogView ) {
+			dialogView.updatePosition();
+		}
+
+		dialog.off( 'change:isOpen', this.updateDialogPositionCallback );
+	}
+
+	/**
+	 * Stores a bound reference to the _updateDialogPosition method, allowing it to be attached and detached from change event.
+	 */
+	public updateDialogPositionCallback = this._updateDialogPosition.bind( this );
+
+	/**
+	 * An event triggered on dialog opening that sets a new position or restores previous values.
+	 */
+	private _updateDialogPosition( _evt: EventInfo, _name: string, isOpen: boolean ): void {
+		if ( isOpen ) {
+			this._setNewDialogPosition();
+		} else {
+			const dialog = this._editor.plugins.get( 'Dialog' );
+			const dialogView = dialog.view;
+
+			if ( dialogView && dialogView.position === null ) {
+				dialogView.position = DialogViewPosition.EDITOR_TOP_SIDE;
+			}
+		}
+	}
+
+	/**
+	 * Adjusts the dialog position to utilize the empty space on the right side of the editable.
+	 * The new dialog position should be on the right side of the fullscreen view with a 30px margin.
+	 * Only dialogs with the position set to "editor-top-side" should have their position changed.
+	 */
+	private _setNewDialogPosition(): void {
+		if ( !this._editor.plugins.has( 'Dialog' ) ) {
+			return;
+		}
+
+		const dialog = this._editor.plugins.get( 'Dialog' );
+		const dialogView = dialog.view!;
+
+		if ( !dialogView || dialogView.position !== DialogViewPosition.EDITOR_TOP_SIDE ) {
+			return;
+		}
+
+		const fullscreenViewContainerRect = new Rect( this._container! ).getVisible();
+		const editorContainerRect = new Rect( document.querySelector( '.ck-fullscreen__editable' ) as HTMLElement ).getVisible();
+		const dialogRect = new Rect( dialogView.element!.querySelector( '.ck-dialog' ) as HTMLElement ).getVisible();
+
+		if ( fullscreenViewContainerRect && editorContainerRect && dialogRect ) {
+			const DIALOG_OFFSET = 28;
+			dialogView.position = null;
+
+			dialogView.moveTo(
+				fullscreenViewContainerRect.left + fullscreenViewContainerRect.width - dialogRect.width - DIALOG_OFFSET,
```

Comment:
We need to take the scrollbar width into account. 
<img width="460" alt="image" src="https://github.com/user-attachments/assets/bdf835d5-a343-4838-980c-411d237f60d1" />